cms/static/js/views/settings/main.js

```diff
@@ -367,12 +367,19 @@ var DetailsView = ValidatingView.extend({
             message = this.post_enrollment_email_elem.val();
         }
 
-        $.post("/settings/send_test_enrollment_email",
+        // Validation for keyword substitution
+        var validation = KeywordValidator.validate_string(message);
+        if (!validation.is_valid) {
+            message = gettext("There are invalid keywords in your email. Please check the following keywords and try again:");
+            message += "\n" + validation.invalid_keywords.join('\n');
+            alert(message);
```

Comment:
This is also used everywhere else there is an alert, should that be a separate PR to make everything explicit? Although I assume it's understood not to redefine alert as another function, so hypothetically it should be fine. But I get what you are saying, better safe than sorry.